### DISCUSSION TITLE
How to support Account-Linking with providers using POST callbacks

### DISCUSSION THREAD

**Discussion by armisael** [nextauthjs/next-auth#6977] [Help]
### Question 💬

I'm implementing a page in our next.js app where authenticated users can see their linked accounts and handle them: they can unlink a service (we simply delete the related Account in our DB) or link one (we trigger the `signIn()` flow, which will result in the account being linked to the authenticated user).

The problem I'm facing arises with some providers, like Apple, that call the OAuth callback endpoint with POST instead of GET. Next-auth's session token has `sameSite: 'lax'`: browsers will send it in incoming requests from third-party websites only on GET; since Apple performs a POST, the cookie is not sent, the session can't be loaded, the user is considered to be anonymous, and a new user is created.

While testing it locally I was able to get around this by setting `sameSite: 'none'` on the session token, but this means the token will be sent on any cross-origin requests, and I'm not sure this is safe enough.

Have you ever stumbled upon a similar issue? Is setting `sameSite: 'none'` safe on the session token? Is there anything else I could do to support account-linking on POST callbacks without diminishing my app security?

### How to reproduce ☕️

To debug whether the session token is available in the callback, just add this in `node_modules/next-auth/core/index.js`, inside `async function AuthHandler`, after `await (0, _init.init)({..})`

```
  if (req.action === 'callback') {
    _logger.default.debug("REQUEST", req);
  }
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
Setting `sameSite: "none"` for the session cookie is not recommended. You likely don't want to send your session to third parties.

Unfortunately, I don't have an Apple device and am not even sure how to get an account to even test our Apple integration... 😬 Even the current implementation was done via communicating with a community-member on Twitter in DMs. :facepalm:

**Reply by armisael** [reply]
Oh wow 🤦 

I tried some other providers I know, looking for one using POST instead of GET, but Apple is the only one I could find; which is not surprising, as the OAuth specifications expect the user to be redirected to the callback, not to be POSTed there...

If changing `sameSite` is not a viable option, and Apple does not allow to configure their flow behaviour, I see only two solutions:
1) in next-auth, detect the request is a POST and redirect to the same endpoint with all parameters in GET; at this point, the browser should add the cookie to the request;
2) give up and disable account linking on Apple in our application.

Do you see any other road we could take?
